app/popups/ServerPasswordPopup.js

```diff
@@ -4,15 +4,33 @@ import * as ActionTypes from '../constants/ActionTypes'
 
 class ServerPasswordPopup extends Component
 {
+    constructor(props)
+    {
+        super(props);
+
+        this.state = {
+            isCapsLockOn: false
+        }
+    }
+
     render()
     {
         return (
             <div className="center-notice">
                 <div className="notice-content">
                     <h1>Enter Server Password</h1>
                     <form onSubmit={this.onSubmit.bind(this)}>
-                        <label htmlFor="password">Server Password</label><br/>
-                        <input type="password" name="password" ref="password" id="password" />
+                        <div className="label-wrapper">
+                            <label htmlFor="password">Server Password</label><br/>
+                            {
+                                this.state.isCapsLockOn ? <div className="caps-lock-notice">
+                                    <i className="material-icons">info</i>CAPS LOCK IS ON
+                                </div>: null
+                            }
+                        </div>
+
+                        <input type="password" name="password" ref="password" id="password" onKeyUp={this.onKeyUp.bind(this)} />

```

Comment:
Use more events here than just `onKeyUp` otherwise this won't update in certain cases. See `containers/Login.js` for some and would also suggest using `onFocus`.

Reply:
Can probably also check it in `componentDidMount`.

Reply:
How to check on componentDidMount? Does it have any params that expose an event with `getModifierState`?

Reply:
I added the events you had in Login (didnt notice you already had an implentation there 😬 )
onFocus doesnt work because that event doesnt have the `getModifierState`.